tests/foreman/cli/test_remoteexecution.py

```diff
@@ -1019,3 +1020,204 @@ def test_positive_rex_against_infra_hosts(
+        try:
+            assert result['success'] == '1'
+        except AssertionError:
+            result = 'host output: {}'.format(
+                ' '.join(
+                    JobInvocation.get_output(
+                        {'id': invocation_command['id'], 'host': rhel_contenthost.hostname}
+                    )
+                )
+            )
+            raise AssertionError(result)
+
+    @pytest.mark.tier3
+    @pytest.mark.upgrade
+    @pytest.mark.rhel_ver_match('[^6].*')
+    def test_positive_run_job_on_host_registered_to_pull_provider(
+        self,
+        module_org,
+        smart_proxy_location,
+        module_ak_with_cv,
+        module_capsule_configured_mqtt,
+        rhel_contenthost,
+    ):
+        """Run custom template on host registered to mqtt
+
+        :id: 759ad51d-eea7-4d7b-b6ee-60af2b814464
+
+        :expectedresults: Verify the job was successfully ran against the host registered to mqtt
+
+        :CaseImportance: Critical
+
+        :parametrized: yes
+        """
+        result = rhel_contenthost.execute(
+            f'curl -o /etc/pki/ca-trust/source/anchors/satellite-sat-engineering-ca.crt \
+                    {settings.repos["DOGFOOD_REPO_HOST"]}/pub/katello-server-ca.crt; \
+                    update-ca-trust'
+        )
+        assert result.status == 0, 'Failed to download certificate'
+        client_repo = (
+            f'{settings.repos["DOGFOOD_REPO_HOST"].replace("http", "https")}/pulp/content/'
+            'Satellite_Engineering/QA/Satellite_Client/custom/Satellite_Client_Composes/'
+            f'Satellite_Client_RHEL{rhel_contenthost.os_version.major}_x86_64/'
+        )
+        # TODO client_repo should be changed to
+        # settings.repos['SATCLIENT_REPO'][f'RHEL{rhel_contenthost.os_version.major}']
+        # when/if the new dogfood url pattern settles
+
+        # register host with pull provider rex (SAT-1677)
+        result = rhel_contenthost.register(
+            module_capsule_configured_mqtt,
+            module_org,
+            smart_proxy_location,
+            module_ak_with_cv.name,
+            setup_remote_execution_pull=True,
+            repo=client_repo,
+        )
+        assert result.status == 0, f'Failed to register host: {result.stderr}'
+        # check mqtt client is running
+        result = rhel_contenthost.execute('yggdrasil status')
+        assert result.status == 0, f'Failed to start yggdrasil on client: {result.stderr}'
+        # run script provider rex command
+        invocation_command = make_job_invocation(
+            {
+                'job-template': 'Service Action - Script Default',
+                'inputs': 'action=status, service=yggdrasild',
+                'search-query': f"name ~ {rhel_contenthost.hostname}",
+            }
+        )
+        result = JobInvocation.info({'id': invocation_command['id']})
+        try:
+            assert result['success'] == '1'
+        except AssertionError:
```

Comment:
This is done about 17 times in this module, might be time to write a function to cover the assertion pattern. Optional for this PR.

Reply:
submitted https://github.com/SatelliteQE/robottelo/issues/9985 for this